pkg/cronjob/spark_dependencies_test.go

```diff
@@ -79,6 +79,17 @@ func TestCreate(t *testing.T) {
 	assert.NotNil(t, CreateSparkDependencies(&v1.Jaeger{Spec: v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Type: "elasticsearch"}}}))
 }
 
+func TestSparkDependenciesSecretSecrets(t *testing.T) {
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestSparkDependenciesSecretSecrets"})
+	secret := "mysecret"
+	jaeger.Spec.Storage.SecretName = secret
+
+	days := 0
+	jaeger.Spec.Storage.EsIndexCleaner.NumberOfDays = &days
+	cronJob := CreateSparkDependencies(jaeger)
+	assert.Equal(t, secret, cronJob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.LocalObjectReference.Name)
```

Comment:
Ideally, there would be two assertions before this one, like:

```
assert.Len(t, cronJob.Spec.JobTemplate.Spec.Template.Spec.Containers, 1)
assert.Len(t, cronJob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].EnvFrom, 1)
```

If some other code causes the Container/EnvFrom list to change, the error message will be less cryptic.

Reply:
I have added those assertions.

I used the test in `es_index_cleaner_test.go` as a reference, so maybe that could have those assertions added too.

Reply:
+1, it should probably be added to all of the tests related to the `secretName` - if you don't mind including them in this PR.